frontend/src/PortfolioInfo.jsx

```diff
@@ -0,0 +1,195 @@
+import { SidebarProvider, SidebarTrigger } from "./components/ui/Sidebar";
+import AppSidebar from "./components/AppSidebar";
+import SearchBar from "./components/SearchBar";
+import { UserInfo } from "./context/UserContext";
+import { useState, useEffect } from "react";
+import { BASE_URL } from "./lib/utils";
+import { useParams, useNavigate } from "react-router-dom";
+import { Pencil, X } from "lucide-react";
+
+const PortfolioInfo = () => {
+  const [portfolioData, setPortfolioData] = useState(null);
+  const [companyIds, setCompanyIds] = useState([]);
+  const [companiesData, setCompaniesData] = useState([]);
+  const [companiesStockData, setCompaniesStockData] = useState([]);
+  const [isEditingMode, setIsEditingMode] = useState(false);
+
+  const navigate = useNavigate();
+
+  const { fullName } = UserInfo();
+  const { id } = useParams();
+
+  const getPortfolioData = async () => {
+    const response = await fetch(`${BASE_URL}/portfolios/${id}`, {
+      method: "GET",
+      credentials: "include",
+    });
+    const data = await response.json();
+    setPortfolioData(data);
+    if (companyIds.length === 0) {
+      setCompanyIds(data.companiesIds);
+    }
+  };
+
+  const handleDelete = async (companyId) => {
+    await fetch(`${BASE_URL}/portfolios/${id}/${companyId}`, {
+      method: "DELETE",
+      credentials: "include",
+    });
+    getCompaniesData();
+    setCompanyIds((self) => self.filter((cid) => cid !== companyId));
+  };
+
+  const getCompaniesData = async () => {
+    const newArray = [];
+    for (let ids of companyIds) {
+      const response = await fetch(`${BASE_URL}/getters/companyById/${ids}`, {
+        method: "GET",
+        credentials: "include",
+      });
+      if (response.ok) {
+        let data = await response.json();
+        newArray.push({ id: data.id, name: data.name, ticker: data.ticker });
+      }
+    }
+    setCompaniesData(newArray);
+    const prices = [];
+    for (const company of newArray) {
+      const stockResponse = await fetch(
```

Comment:
Same feedback: any way to batch the network requests as a single one, and then unpack them into individual API calls if needed? I'm guessing if we just query the DB on the backend that should be doable as a single operation.